app/src/main/java/ch/epfl/sdp/item/Trap.java

```diff
@@ -0,0 +1,72 @@
+package ch.epfl.sdp.item;
+
+import ch.epfl.sdp.artificial_intelligence.Updatable;
+import ch.epfl.sdp.entity.EntityType;
+import ch.epfl.sdp.entity.Player;
+import ch.epfl.sdp.entity.PlayerManager;
+import ch.epfl.sdp.game.Game;
+import ch.epfl.sdp.map.Displayable;
+import ch.epfl.sdp.map.GeoPoint;
+
+/**
+ * A trap is an item a player can drop at his current localization and if another player walks by the trap, he will take damage
+ */
+public class Trap extends Item implements Updatable, Displayable {
+    private GeoPoint trapPosition;
+    private int damage;
+    private int radius;
+    private Player owner;
+
+    /**
+     * This is a constructor for traps
+     * @param damage The damage the trap will inflict on the other player that walks by
+     * @param radius The radius of the circle in which the other player needs to be in order for the trap to activate
+     */
+    public Trap(int damage, int radius) {
+        super(String.format("Trap (%d)", damage), String.format("Deal %d health points damages to the player that walks on it", damage));
+        this.damage = damage;
+        this.radius = radius;
+    }
+
+    @Override
+    public void use() {
+        //This is called by the player that has the item, so getUser should return the correct player
+        //The trick is that we need to save the reference to the player, since the update method will be called by the server and thus,
+        //getUser would return the wrong player (the server)

```

Comment:
I don't see why PlayerManager.getUser() will return different users at different instants. Maybe you can explain this in the comments.

Reply:
It will return a different user in each phone (Client).

Reply:
Thus, only the phone that called use should be save in the trap memory and we cannot simply use the getUser method when we try to find which player should get damages